app/src/main/java/net/squanchy/eventdetails/EventDetailsService.java

```diff
@@ -0,0 +1,44 @@
+package net.squanchy.eventdetails;
+
+import net.squanchy.schedule.domain.view.Event;
+import net.squanchy.service.firebase.FirebaseSquanchyRepository;
+import net.squanchy.service.firebase.model.FirebaseEvent;
+import net.squanchy.service.firebase.model.FirebaseSpeaker;
+
+import io.reactivex.Observable;
+import io.reactivex.schedulers.Schedulers;
+
+import static net.squanchy.support.lang.Lists.find;
+import static net.squanchy.support.lang.Lists.map;
+
+class EventDetailsService {
+
+    private final FirebaseSquanchyRepository repository;
+
+    EventDetailsService(FirebaseSquanchyRepository repository) {
+        this.repository = repository;
+    }
+
+    public Observable<Event> event(int dayId, int eventId) {
+        Observable<FirebaseEvent> eventObservable = repository.event(dayId, eventId);
+        Observable<FirebaseSpeaker.Holder> speakersObservable = repository.speakers();
+
+        return Observable.combineLatest(
+                eventObservable,
+                speakersObservable,
+                (firebaseEvent, speakerHolder) -> Event.create(
```

Comment:
Change of direction again, will rename models too, so it's a big change, separate PR incoming as soon as this is shipped